aten/src/ATen/native/Distributions.cpp

```diff
@@ -296,25 +296,36 @@ Tensor _s_dirichlet_cpu(const Tensor& alpha, Generator *gen) {
   return ret;
 }
 
-Tensor multinomial_cpu(const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
-  Tensor result = at::empty({0}, self.options().dtype(kLong));
-  multinomial_out_cpu(result, self, n_sample, with_replacement, gen);
-  return result;
-}
-
-Tensor& multinomial_out_cpu(Tensor& result, const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
-  TORCH_CHECK(at::isFloatingType(self.scalar_type()), "multinomial only supports floating-point dtypes for input, got: ", self.scalar_type());
-  TORCH_CHECK(result.scalar_type() == ScalarType::Long, "multinomial expects Long tensor out, got: ", result.scalar_type());
+/* The largest consecutive integer representable in float32 (2^24) */
+constexpr int64_t FLOAT32_MAX_CONSECUTIVE_INT = 16777216;
```

Comment:
isn't this just `(1 << (FLT_MANT_DIG))` ?

Reply:
yes it's

Reply:
I agree with @gchanan, this line would be nicer as 1 << (FLT_MANT_DIG).